### PR TITLE
chore: update publish to occur after release

### PR DESCRIPTION
**Pull request by gbockus-sf**
### What does this PR do?
- Update the current flow to create a tag & release on merge with main.
- Move the ctc jobs & add a publish job to be triggered a new release is triggered. 
This will allow us to download the vsix files from the release and test prior to publishing to the vscode Marketplace. 

### What issues does this PR fix or reference?
@W-11596486@

### Functionality Before
Unable to get access to vsix files prior to publishing to vscode marketplace

### Functionality After
Merge With Main -> Trigger tag and release in github -> Trigger publish to vscode marketplace

### CODE REVIEW

File: .github/workflows/prerelease.yml
Comment (gbockus-sf) [inline .github/workflows/prerelease.yml:2]: This now triggers a release so I named it appropriately.

File: .github/workflows/publishVSCode.yml
Comment (gbockus-sf) [inline .github/workflows/publishVSCode.yml:21]: Not this uses the publish env so it will wait for approval.

File: .github/workflows/publishVSCode.yml
Comment (gbockus-sf) [inline .github/workflows/publishVSCode.yml:70]: Using the gh cli to download the vsix files attached to the release which are generated in the release workflow.

File: .github/workflows/release.yml
Comment (gbockus-sf) [inline .github/workflows/release.yml:37]: This is taken from the old publish flow.

File: .github/workflows/tagAndReleaseVSCode.yml
Comment (gbockus-sf) [inline .github/workflows/tagAndReleaseVSCode.yml:109]: same story..just changing names from publish to release in this flow.

File: .github/workflows/publishVSCode.yml
Comment (randi274) [inline .github/workflows/publishVSCode.yml:13]: Nit: single quotes for consistency

File: .github/workflows/publishVSCode.yml
Comment (randi274) [reply .github/workflows/publishVSCode.yml:70]: Ohhh I like this

File: .github/workflows/publishVSCode.yml
Comment (gbockus-sf) [reply .github/workflows/publishVSCode.yml:13]: uh...actually it looks like all these slack calls use double quotes though...I'll update the name for consistency

**Review by randi274 — APPROVED** [review]

**Pull request merged**
